app/src/main/java/com/bogdan/codeforceswatcher/features/main/BottomNavigationMenuFragment.kt

```diff
@@ -0,0 +1,117 @@
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.fragment.app.Fragment
+import com.bogdan.codeforceswatcher.R
+import com.bogdan.codeforceswatcher.components.compose.theme.AlgoismeTheme
+
+class NavigationMenuFragment: Fragment() {
+
+    private val items = listOf(
+        MenuItem(MenuItem.Tab.CONTESTS, R.drawable.ic_contests, R.drawable.ic_contests_gradient),
+        MenuItem(MenuItem.Tab.USERS, R.drawable.ic_users, R.drawable.ic_users_gradient),
+        MenuItem(MenuItem.Tab.NEWS, R.drawable.ic_news, R.drawable.ic_news_gradient),
+        MenuItem(MenuItem.Tab.PROBLEMS, R.drawable.ic_problems, R.drawable.ic_problems_gradient),
+    )
+    private var selectedTab: MutableState<MenuItem.Tab> = mutableStateOf(MenuItem.Tab.CONTESTS)
+    lateinit var onMenuItemTap: (MenuItem.Tab) -> Unit
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ) = ComposeView(requireContext()).apply {
+        setContent {
+            AlgoismeTheme {
+                NavigationMenuView(items, selectedTab) {
+                    selectedTab.value = it
+                    onMenuItemTap(it)
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun NavigationMenuView(
+    items: List<MenuItem>,
+    selectedTab: State<MenuItem.Tab>,
+    onMenuItemTap: (MenuItem.Tab) -> Unit
+) = Row(
+    modifier = Modifier
+        .clip(RoundedCornerShape(topStart = 30.dp, topEnd = 30.dp))
+        .background(AlgoismeTheme.colors.primaryVariant)
+        .padding(horizontal = 14.dp),
+    horizontalArrangement = Arrangement.SpaceBetween,
+    verticalAlignment = Alignment.CenterVertically
+) {
+    items.subList(0, 2).forEach {
+        NavigationMenuItemView(it, selectedTab.value, onMenuItemTap, Modifier.weight(1f))
+    }
+
+    Spacer(modifier = Modifier.fillMaxWidth(0.15f))
+
+    items.subList(2, 4).forEach {
+        NavigationMenuItemView(it, selectedTab.value, onMenuItemTap, Modifier.weight(1f))
+    }
+}
+
+@Composable
+private fun NavigationMenuItemView(
+    item: MenuItem,
+    selectedTab: MenuItem.Tab,
+    onMenuItemTap: (MenuItem.Tab) -> Unit,
+    modifier: Modifier = Modifier
+) = Column(
+    modifier = modifier.clickable { onMenuItemTap(item.tab) },
+    horizontalAlignment = Alignment.CenterHorizontally,
+    verticalArrangement = Arrangement.spacedBy(2.dp)
+) {
+    Image(
+        painter = painterResource(if (item.tab == selectedTab) item.selectedIconId else item.iconId),
+        contentDescription = null,
+        colorFilter = if (item.tab == selectedTab) null else ColorFilter.tint(AlgoismeTheme.colors.secondaryVariant)
+    )
+
+    Text(
+        text = stringResource(tabTitle(item.tab)),
+        style = AlgoismeTheme.typography.hintRegular.copy(fontSize = 11.sp),
+        color = AlgoismeTheme.colors.secondaryVariant
+    )
+}
+
+private fun tabTitle(tab: MenuItem.Tab) = when(tab) {
```

Comment:
More Kotlin way would be `private val MenuItem.Tab.title get() = when (this) {}`.